code/go/0chain.net/blobbercore/stats/dbstats.go

```diff
@@ -0,0 +1,88 @@
+package stats
+
+import (
+	"database/sql"
+	"encoding/json"
+	"github.com/0chain/blobber/code/go/0chain.net/blobbercore/datastore"
+	"github.com/0chain/blobber/code/go/0chain.net/core/common"
+	"log"
+	"time"
+)
+
+// customDBStats contains database statistics.
+type customDBStats struct {
+	MaxOpenConnections int // Maximum number of open connections to the database.
+
+	// Pool Status
+	OpenConnections int // The number of established connections both in use and idle.
+	InUse           int // The number of connections currently in use.
+	Idle            int // The number of idle connections.
+
+	// Counters
+	WaitCount         int64         // The total number of connections waited for.
+	WaitDuration      time.Duration // The total time blocked waiting for a new connection.
+	MaxIdleClosed     int64         // The total number of connections closed due to SetMaxIdleConns.
+	MaxIdleTimeClosed int64         // The total number of connections closed due to SetConnMaxIdleTime.
+	MaxLifetimeClosed int64         // The total number of connections closed due to SetConnMaxLifetime.
+	Status            string
+}
+
+var myCustomDBStats *customDBStats
+
+func init() {
+	myCustomDBStats = &customDBStats{Status: "connecting"}
+}
+
+func SetDBStatStatusFail() {
+	myCustomDBStats = &customDBStats{Status: "✗"}
+}
+
+func SetDBStatStatusOK() {
+	myCustomDBStats.Status = "✔"
+}
+
+func ConvertSqlDBStatsToCustomDBStats(in *sql.DBStats) error {
+	err := toSubStruct(&in, myCustomDBStats)
+	if err != nil {
+		return common.NewErrorf("struct_copy_error", "Error copying struct: %v", err)
+	}
+
+	return nil
+}
+
+func toSubStruct(in, out interface{}) error {
```

Comment:
in my opinion, it is unnecessary to extract them into a standalone method. Because `ConvertSqlDBStatsToCustomDBStats` is enough simple now.

Reply:
the same issue, there are too many small methods. code has too many pice for a small feature.

Reply:
it is better to  follow `Single Responsibility` to design your method

Reply:
okay, done.